File: backend/app/models/torneo.py
from sqlalchemy import Column, Integer, String, Boolean, ForeignKey, Date
from sqlalchemy.orm import relationship

from app.models.base import Base, TimestampMixin


class Torneo(Base, TimestampMixin):
    __tablename__ = "torneos"

    id = Column(Integer, primary_key=True, index=True)
    nombre = Column(String, nullable=False)
    descripcion = Column(String, nullable=True)
    deporte = Column(String, nullable=True)
    modalidad = Column(String, nullable=True)
    direccion = Column(String, nullable=True)
    fecha_inicio = Column(Date, nullable=True)
    fecha_fin = Column(Date, nullable=True)
    estado = Column(String, default="Pendiente")
    imagen_url = Column(String, nullable=True)
    activo = Column(Boolean, nullable=False, default=True)

    user_id = Column(Integer, ForeignKey("users.id"), nullable=True)

    usuario = relationship("User", back_populates="torneos")
    equipos = relationship("Equipo", back_populates="torneo", cascade="all, delete-orphan")
    partidos = relationship("Partido", back_populates="torneo", cascade="all, delete-orphan")

File: mobile/src/types/torneo.ts
export type Torneo = {
  id: number;
  user_id?: number | null;
  nombre: string;
  deporte: string;
  modalidad: string;
  descripcion?: string | null;
  direccion?: string | null;
  fecha_inicio?: string | null;
  fecha_fin?: string | null;
  estado: string;
  imagen_url?: string | null;
  created_at?: string;
  updated_at?: string;
};